libraries/network/src/main/kotlin/io/element/android/libraries/network/RetrofitFactory.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.network
+
+import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
+import io.element.android.libraries.core.uri.ensureTrailingSlash
+import kotlinx.serialization.json.Json
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import javax.inject.Inject
+
+class RetrofitFactory @Inject constructor(
+    private val okHttpClient: OkHttpClient,
```

Comment:
Inject Lazy<OkHttpClient> and then use callFactory instead of setting client manually. It'll allow to initialise retrofit/okhttp lazily

```
fun create(baseUrl: String): Retrofit {
        val contentType = "application/json".toMediaType()
        return Retrofit.Builder()
            .baseUrl(baseUrl.ensureTrailingSlash())
            .addConverterFactory(Json.asConverterFactory(contentType))
            .callFactory(object : Call.Factory {
                    override fun newCall(request: Request): Call {
                        return okHttpClient.get().newCall(request)
                    }
                })
            .build()
    }
```

Reply:
728a19c0fb76f54b4cf9abd6fa0661fc8ad47b82